syntax = "proto3";
service Movie {
  rpc GetMovieByID(MovieID) returns (MovieData) {}
  rpc GetListMovies(Empty) returns (stream MovieData) {}
  rpc CreateMovie(MovieData) returns (Result) {}
  rpc UpdateMovieRating(MovieRating) returns (Result) {}
  rpc DeleteMovie(MovieID) returns (Result) {}

}

message MovieRating {
  string id = 1;
  float rating = 2;
}

message MovieID {
  string id = 1;
}
message MovieData {
  string title = 1;
  float rating = 2;
  string director = 3;
  string id = 4;
}
message Empty {
}

message Result {
  string message= 1;
}